code/GlobalNavControllerExtension.php

```diff
@@ -6,7 +6,7 @@ public function onBeforeInit() {
 		$host = GlobalNavSiteTreeExtension::get_toolbar_hostname();
 		if (
 				(isset($_REQUEST['regenerate_nav']) || isset($_REQUEST['flush'])) && 
-				$host == Director::absoluteBaseURL() && 
```

Comment:
I can't see why this would have failed unless GlobalNav.use_localhost was set and GlobalNav.hostname was not equal to the base url, which is set by config right?